### PR TITLE
(FACT-1384) Remove deprecated API use required by Server 2003

### PR DESCRIPTION
**Pull request by MikaelSmith**
Switches to using SHGetKnownFolderPath via a new Leatherman helper.
Removes use of the deprecated SHGetFolderPath.

Removes use of old APIs for networking. Also bumps Leatherman to 0.9.0.

### CODE REVIEW

**Comment by Iristyle** [comment]
merge conflicts @MikaelSmith

**Comment by MikaelSmith** [comment]
Fixed.

File: appveyor.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: 3.1.0.{build}
 clone_depth: 10
 environment:
-  LEATHERMAN_VERSION: 0.8.1
+  LEATHERMAN_VERSION: 0.9.0
```
Comment (Magisus) [inline appveyor.yml]: Same here, should probably be 0.9.1

**Comment by MikaelSmith** [comment]
Updated.

**Comment by branan** [comment]
puppet-agent stable is still on Leatherman 0.8.1. This PR will need to target master.

**Comment by branan** [comment]
@MikaelSmith I rebased this to master and there are new and exciting conflicts

**Comment by MikaelSmith** [comment]
Updated.

File: lib/src/facts/windows/networking_resolver.cc [hunk 2]
```diff
@@ -26,13 +24,6 @@ namespace facter { namespace facts { namespace windows {
 
     networking_resolver::networking_resolver()
     {
-        // Find ConvertLengthToIpv4Mask and save it to _convertLengthToIpv4Mask. Won't be found on
-        // Windows Server 2003, but in 2003 we get masks as strings so this function isn't needed.
-        auto func = GetProcAddress(GetModuleHandleW(L"Iphlpapi"), "ConvertLengthToIpv4Mask");
-        if (nullptr != func) {
-            typedef int (WINAPI *LPFN_CONVLEN) (ULONG, PULONG);
-            _convertLengthToIpv4Mask = reinterpret_cast<LPFN_CONVLEN>(func);
```
Comment (branan) [inline lib/src/facts/windows/networking_resolver.cc]: Where is this variable declared? I don't see it being removed here

Comment (MikaelSmith) [inline lib/src/facts/windows/networking_resolver.cc]: Removed from the header file now as well.

**Comment by MikaelSmith** [comment]
Missed it in tests, trying again.

**Comment by Magisus** [comment]
@branan are you good on this now that Appveyor has passed?

**Pull request merged**